### PR TITLE
Allow aeson >= 2.0 and fix missing Lift instance

### PR DESCRIPTION
**Pull request by mmhat**
This pull requests contains two things:

 * It lifts the restriction on the `aeson` dependency. To achieve that it adds a utility function `fromAesonMap` that is defined using CPP to work with the aeson version used.
 * It builds with containers >= 0.6.6. Apparently the orphan `Lift` instance we used from the string-interpolate package is not present when built against containers ==0.6.6 as the instance is part of that package now.

In both cases the compatibility code was added to a new internal module `Grace.Compat`. With this patch Grace is buildable with GHC 8.10.7, 9.0.2, 9.2.4 and 9.4.2. The tasty testsuite passes for all of them except 9.4.2; For this GHC version Cabal is unable to produce a build plan.

### CODE REVIEW

**Comment by Gabriella439** [comment]
I have two broad suggestions:

- I don't think we need to support older versions of `containers`

  Generally, for `grace` backwards compatibility is not that important unless we need it for the GHCJS build.  I think `aeson` is the only package where we need to support an older version for GHCJS

- Perhaps use Cabal's support conditional compilation instead of `CPP`

  Specifically, ,I'm thinking of the approach outlined here: http://blog.haskell-exists.com/yuras/posts/stop-abusing-cpp-in-haskell.html

  In other words, create an `aeson-1/` and `aeson-2/` directory that Cabal chooses between depending on the `aeson` version.

**Comment by mmhat** [comment]
> I have two broad suggestions:
> 
> * I don't think we need to support older versions of `containers`
>   Generally, for `grace` backwards compatibility is not that important unless we need it for the GHCJS build.  I think `aeson` is the only package where we need to support an older version for GHCJS

Well, trying `cabal test tasty --constraint='containers>=0.6.6'` results in Cabal not being able to construct a build plan. So I guess we have to support a lower constainers version for the sake of our teststuite.

> * Perhaps use Cabal's support conditional compilation instead of `CPP`
>   Specifically, ,I'm thinking of the approach outlined here: http://blog.haskell-exists.com/yuras/posts/stop-abusing-cpp-in-haskell.html
>   In other words, create an `aeson-1/` and `aeson-2/` directory that Cabal chooses between depending on the `aeson` version.

I don't think that will work as intended. I am not aware of conditionals like
```cabal
if foo(aeson>=2.0)
then
  hs-source-dirs: aeson-2/
else
  hs-source-dirs: aeson-1/
```
in Cabal. So either we drop support for aeson<2.0 or we go with CPP (I don't like it either).

**Comment by Gabriella439** [comment]
Ah, good points.  Never mind, then! 🙂
